### ISSUE TITLE
Speed up reading of single variants?

### ISSUE THREAD

**Issue by koelling** [limix/pandas-plink#17]
Hi Danilo,

Thanks for making this package, it's very useful!

One relatively common use case for me is that I have a big PLINK file and want to get a data frame with the genotypes for a single variant. To do that I run something like this:

```
G = read_plink1_bin('file.bed', verbose=False)
var = G.sel(variant='{}_{}'.format(chrom, snp_id))
gt = pd.DataFrame({'sample': var.sample.values, 'gt': var.values})
```

This does work, but if the input file is big it takes surprisingly long. In particular the `read_plink1_bin` step is very slow. I assume this is because it's reading quite a lot of data about all the other variants into memory, which is never actually used.

Is there a way to tell `read_plink1_bin` that I only care about a single variant in the file and ignore the rest?

**Comment by koelling** [comment]
And on a related note - is there a way to query a single variant based only on its ID, without knowing the chromosome it's on?

**Comment by horta** [comment]
Hi @koelling Sorry for the long delay. I have just released a new version (2.1.0) with performance improvement and allowing the user to fine tune the chunking (which might help you a lot). Doc: https://pandas-plink.readthedocs.io/en/latest/api/pandas_plink.read_plink1_bin.html


Regarding selecting single variant based on its ID, there is. Instead of

```python
import pandas as pd

from pandas_plink import read_plink1_bin

name = "ALL.chr1.phase1_release_v3.20101123.snps_indels_svs.genotypes.bed"
G = read_plink1_bin(name)
v = G.sel(variant="1_rs202225383")
gt = pd.DataFrame({"sample": v.sample.values, "gt": v.values})
print(gt)
```

you can do

```python
import pandas as pd

from pandas_plink import read_plink1_bin

name = "ALL.chr1.phase1_release_v3.20101123.snps_indels_svs.genotypes.bed"
G = read_plink1_bin(name)
v = G.where(G.snp == "rs202225383", drop=True)
v = v.squeeze()
gt = pd.DataFrame({"sample": v.sample.values, "gt": v.values})
print(gt)
```

Let me know if you still find it slow. Closing it for now.

**Comment by koelling** [comment]
Thank you Danilo, will give it a try!
